### ISSUE TITLE
setup.py try-except block hides fortran compilation problems

### ISSUE THREAD

**Issue by nmusolino** [andim/scipydirect#12]
The project's README.md file says:

> If you receive the warning message "Fortran code not compiled", when you try to use the package the installation has not been successful. Numpy.distutils is used to handle the Fortran dependency, so you might find advice for how to solve your problems by searching for "numpy distutils fortran" online. Unfortunately Fortran installation issues seem to be common, but I cannot assist you in that regard as such an issue is specific to the details of your computer setup.

The current `setup.py` file suppresses exceptions from extension module compilation, and modern pip (e.g. `pip 22.0.4`, used below) does not show the warning message printed in `setup.py`:

```
    try:
        thiskwargs = kwargs.copy()
        config = configuration()
        config.add_extension('direct', sources=['src/direct.pyf', 'src/DIRect.f', 'src/DIRserial.f', 'src/DIRsubrout.f'])
        thiskwargs.update(config.todict())
        setup(**thiskwargs)
    except:
        # if there was an error try building module without Fortran extension
        # the module will not be usable, but documentation can be built
        # (for readthedocs)
        warnings.warn('There was an error with building the Fortran extension.')
        thiskwargs = kwargs.copy()
        config = configuration()
        thiskwargs.update(config.todict())
        setup(**thiskwargs)
```

```
$ pip install -e .
[...]
Installing collected packages: scipydirect
  Attempting uninstall: scipydirect
    Found existing installation: scipydirect 1.3
    Uninstalling scipydirect-1.3:
      Successfully uninstalled scipydirect-1.3
  Running setup.py develop for scipydirect
Successfully installed scipydirect-1.3
```

This output was obtained in an environment _without_ a Fortran compiler, and the resulting package was not functional:
```
$ python3.8 -c 'import scipydirect'
Fortran code not compiled, module not functional
```

If the `scipydirect` extension module cannot be built, it is much more conventional and useful to fail at installation time, rather than to silently succeed, only to later print a message at module load time.  Installing a non-functional module for the docs is generally not useful to users.

**Comment by andim** [comment]
Thanks for your note!.

The reason for failing silently isn't to allow users to generate docs for a non-functional module (which I agree isn't very useful). Instead, as you can see in the comment in the code the reason for silent failure is to enable compiling the documentation you can find at https://scipydirect.readthedocs.io/

Unfortunately at the time I was developing the module read the docs did not provide a Fortran compiler. (I am not sure whether that has changed.) If you are aware of other ways of solving this compilation issue, please tell me and I'd be happy to update the module.

**Comment by andim** [comment]
This is fixed by dd4f51f8277b1132e8bb8729dc9871da2e48935b. I have updated the read the docs integration to correctly install the package by requiring prior installation of gfortran.

**Comment by nmusolino** [comment]
Thanks!  I think the changes in dd4f51f are the right thing the do, and hopefully will lead to less user confusion.  Thanks again.
